src/typechecker.erl

```diff
@@ -1612,9 +1605,11 @@ do_type_check_expr(Env, {call, _, {atom, _, TypeOp}, [Expr, {string, _, TypeStr}
     end;
 do_type_check_expr(Env, {call, _, {atom, _, record_info}, [_, _]} = Call) ->
     Ty = get_record_info_type(Call, Env),
-    {Ty, #{}, constraints:empty()};
+    {Ty, Env, constraints:empty()};
```

Comment:
This response is also connected with your other one from above, so I'll respond to both here:

> What I had in mind for disabling exhaustiveness-checking was merely to add another element to the return tuple of add_type_pat (i.e. {_, _, _, _, DisableExhaust}).

> Arguably, it's a more functional style to pass as little data as possible to and from each function.

The thing is this value is not used immediately after return from `add_type_pat`. Possibly, it could even be only used after return from an N-level nested `add_type_pat` call, not the topmost one. Then, it has to be returned up back to `check_clauses`. Would it make sense to modify each of these to return the 5th tuple element? I think that with the current change, even though it is bigger in scope, the code is more future-proof, as whatever we might need to store/change in `Env` as we type check will be available in all the subsequent code.

Reply:
Right, it affects all outer patterns! I didn't think of that.

Why does it affect expressions though?

Reply:
> Then, it has to be returned up back to `check_clauses`.

How do we distinguish between levels here? We have exhaustiveness-checking not only for function cases but also for case, receive and try clauses. We want to disable it only for the current "clause-related" expression, not for e.g. other (inner or outer) case expressions. Or am I missing something?